### PR TITLE
New: Added support for Gotify notifications

### PR DESCRIPTION
**Pull request by stephanrenggli**
#### Database Migration
NO

#### Description
This adds support for Gotify (self-hosted notification service similar to pushover) to Sonarr.

#### Todos
- [ ] Tests
- [ ] Documentation

#### Issues Fixed or Closed by this PR

* https://github.com/Sonarr/Sonarr/issues/3033

### CODE REVIEW

File: src/NzbDrone.Core/Notifications/Gotify/GotifyService.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        private readonly Logger _logger;
+
+        public GotifyProxy(Logger logger)
+        {
+            _logger = logger;
+        }
+
+        public void SendNotification(string title, string message, GotifySettings settings)
+        {
+            var URL = settings.GotifyServer + "/message?token=" + settings.AppToken;
```
Comment (markus101) [inline src/NzbDrone.Core/Notifications/Gotify/GotifyService.cs]: Variables should be camel case, `url` not `URL`. Use `string.Format` better yet string interpolation.

File: src/NzbDrone.Core/Notifications/Gotify/GotifyService.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            {
+                const string title = "Test Notification";
+                const string body = "This is a test message from Sonarr";
+
+                SendNotification(title, body, settings);
+            }
+            catch (Exception ex)
+            {
+                _logger.Error(ex, "Unable to send test message");
+                return new ValidationFailure("ApiKey", "Unable to send test message");
```
Comment (markus101) [inline src/NzbDrone.Core/Notifications/Gotify/GotifyService.cs]: There isn't an `ApiKey` setting, but unless the error is specific to an invalid key it should either be on the server or no property.

File: src/NzbDrone.Core/Notifications/Gotify/GotifyService.cs (hunk 1 above)
Comment (stephanrenggli) [reply src/NzbDrone.Core/Notifications/Gotify/GotifyService.cs]: Changed this to use string interpolation to build the url.

File: src/NzbDrone.Core/Notifications/Gotify/GotifyService.cs (hunk 2 above)
Comment (stephanrenggli) [reply src/NzbDrone.Core/Notifications/Gotify/GotifyService.cs]: Removed the `ApiKey` property.

**Comment by markus101** [comment]
Thanks for getting this started, I made some changes and landed this in https://github.com/Sonarr/Sonarr/commit/90626e353f4625798e68ac066f5038d8d4eaf5e5

Part of the cleanup was getting rid of the merge commit, but also fixed the naming of the proxy, removed test from the proxy and renamed a few things.

**Pull request closed without merging**